Add infocom search options to DC assets

cconard96 opened the pull request:
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #9021

The Infocom tab exists for these items already but they aren't in the `infocom_types` array so they don't have the needed search options.

trasher commented:
Your changes are OK, but duplicated SO has to be fixed :/

trasher commented on inc/enclosure.class.php:
```diff
@@ -312,7 +312,7 @@ function rawSearchOptions() {
       ];
 
       $tab[] = [
-         'id'                 => '50',
```
Since this SO already exists in a stable GLPI release, I guess we'll have to add a migration script (for display preferences, and maybe saved searches)

cconard96 commented:
I added new methods to the Migration class to handle search option changes, added tests for those methods, and added the migrations to a new update file.

trasher commented:
Newly added tests are failing; I do not know exactly what's wrong
